schemas/stat_status.fbs

```diff
@@ -0,0 +1,29 @@
+// ATTENTION
+//
+// This schema is still under development and may still break in the future.
+// Backwards compatibility is so far not guaranteed.
+// dominik.werder@gmail.com
+
+
+// Status updates from components in the system.  The actual status information
+// is provided as json, but it is useful to have a top-level flatbuffer for
+// more efficient filtering without having to parse the full json, and also to
+// statically avoid name collisions.
+
+include "cid0_component_id.fbs";
+
+file_identifier "stat";
+
+table Status {
+	// Origin of this status update. Defined in schema 'cid0'.
+	component_id: ComponentID;
+
+	// For simplicity, let's use nanoseconds since unix epoch as e.g. the event
+	// data does as well.
+	timestamp: ulong;
+
+	// The actual status data as json
+	json: string;
```

Comment:
Why JSON here? What does NICOS do with it?

Reply:
The idea is not to abandon json completely, just to move the important information that consumers want to filter on into the flatbuffer. So that json field would then contain the actual service-specific status information. Moving completely to flatbuffer without any embedded json is possible as well of course, but seemed a bit like a too rigid approach to me.

Reply:
If that makes sense from a NICOS perspective that is fine.
If the intention was to make this format suitable for logs in general, if we moved away from Graylog for example, then I think it doesn't make sense. That's because there would be big benefits for the _entire_ message to be JSON in that case. For example existing tools could be used to dump the messages directly into Elasticsearch where all the fields would then be accessible for filtering searches.

Reply:
Yes, for consumers which want to process all messages anyway, it would be nicer to have pure json messages. Other consumers have to parse all json first in order to filter.
And what about the concern that we discussed that we need to separate json topics from flatbuffer topics?

Reply:
It's only a concern if we do decide to send log messages over Kafka.
We are setting up our production Kafka servers right now and I'll be running Graylog, a Kafka metrics reporter and Grafana, so I'll give this more thought soon.